l1-contracts/test/foundry/unit/concrete/Bridges/L1SharedBridge/L1SharedBridgeFails.t.sol

```diff
@@ -14,9 +14,165 @@ import {IMailbox} from "contracts/state-transition/chain-interfaces/IMailbox.sol
+
+        vm.prank(owner);
+        vm.expectRevert("ShB: eFPUB already set");
+        sharedBridge.setEraPostDiamondUpgradeFirstBatch(eraPostUpgradeFirstBatch);
+    }
+
+    function setEraPostLegacyBridgeUpgradeFirstBatch_wrongValue(uint256 eraPostLegacyBridgeUpgradeFirstBatch) public {
+        eraPostUpgradeFirstBatch = bound(eraPostLegacyBridgeUpgradeFirstBatch, 1, type(uint256).max);
+
+        vm.prank(owner);
+        sharedBridge.setEraPostLegacyBridgeUpgradeFirstBatch(eraPostLegacyBridgeUpgradeFirstBatch);
+
+        vm.prank(owner);
+        vm.expectRevert("ShB: eFPUB already set");
+        sharedBridge.setEraPostLegacyBridgeUpgradeFirstBatch(eraPostLegacyBridgeUpgradeFirstBatch);
+    }
+
+    function test_setEraLegacyBridgeLastDepositTime_batchAlreadySet(
+        uint256 eraLegacyBridgeLastDepositBatch,
+        uint256 eraLegacyBridgeLastDepositTxNumber
+    ) public {
+        L1SharedBridge sharedBridgeImpl = new L1SharedBridge({
+            _l1WethAddress: l1WethAddress,
+            _bridgehub: IBridgehub(bridgehubAddress),
+            _eraChainId: eraChainId,
+            _eraDiamondProxy: eraDiamondProxy
+        });
+        TransparentUpgradeableProxy sharedBridgeProxy = new TransparentUpgradeableProxy(
+            address(sharedBridgeImpl),
+            admin,
+            abi.encodeWithSelector(L1SharedBridge.initialize.selector, owner)
+        );
+
+        L1SharedBridge sharedBridge = L1SharedBridge(payable(sharedBridgeProxy));
+
+        vm.startPrank(owner);
+        sharedBridge.setEraLegacyBridgeLastDepositTime(0, 0);
+
+        eraLegacyBridgeLastDepositBatch = bound(eraLegacyBridgeLastDepositTxNumber, 1, type(uint256).max);
+        eraLegacyBridgeLastDepositTxNumber = 0;
+
+        vm.startPrank(owner);
+        sharedBridge.setEraLegacyBridgeLastDepositTime(
+            eraLegacyBridgeLastDepositBatch,
+            eraLegacyBridgeLastDepositTxNumber
+        );
+
+        vm.expectRevert("ShB: eLOBDB already set");
+        sharedBridge.setEraLegacyBridgeLastDepositTime(
+            eraLegacyBridgeLastDepositBatch,
+            eraLegacyBridgeLastDepositTxNumber
+        );
+
+        vm.stopPrank();
+    }
+
+    function test_setEraLegacyBridgeLastDepositTime_txnSet(
+        uint256 eraLegacyBridgeLastDepositBatch,
+        uint256 eraLegacyBridgeLastDepositTxNumber
+    ) public {
+        L1SharedBridge sharedBridgeImpl = new L1SharedBridge({
+            _l1WethAddress: l1WethAddress,
+            _bridgehub: IBridgehub(bridgehubAddress),
+            _eraChainId: eraChainId,
+            _eraDiamondProxy: eraDiamondProxy
+        });
+        TransparentUpgradeableProxy sharedBridgeProxy = new TransparentUpgradeableProxy(
+            address(sharedBridgeImpl),
+            admin,
+            abi.encodeWithSelector(L1SharedBridge.initialize.selector, owner)
+        );
+
+        L1SharedBridge sharedBridge = L1SharedBridge(payable(sharedBridgeProxy));
```

Comment:
same as https://github.com/neotheprogramist/era-contracts/pull/19#discussion_r1633254637

Reply:
This was indeed redundant, nice catch :-)